subsys/net/lib/sntp/sntp.c

```diff
@@ -164,27 +204,30 @@ int sntp_init(struct sntp_ctx *ctx, struct sockaddr *addr, socklen_t addr_len)
 	}
 
 	ctx->sock.fds[ctx->sock.nfds].fd = ctx->sock.fd;
-	ctx->sock.fds[ctx->sock.nfds].events = ZSOCK_POLLIN;
+	ctx->sock.fds[ctx->sock.nfds].events = POLLIN;
 	ctx->sock.nfds++;
 
 	return 0;
 }
 
 int sntp_query(struct sntp_ctx *ctx, uint32_t timeout, struct sntp_time *time)
 {
-	struct sntp_pkt tx_pkt = { 0 };
+	struct sntp_pkt tx_pkt = {0};
 	int ret = 0;
 
 	if (!ctx || !time) {
 		return -EFAULT;
 	}
 
 	/* prepare request pkt */
-	SNTP_SET_LI(tx_pkt.lvm, 0);
-	SNTP_SET_VN(tx_pkt.lvm, SNTP_VERSION_NUMBER);
-	SNTP_SET_MODE(tx_pkt.lvm, SNTP_MODE_CLIENT);
-	ctx->expected_orig_ts = get_uptime_in_sec() + OFFSET_1970_JAN_1;
-	tx_pkt.tx_tm_s = htonl(ctx->expected_orig_ts);
+	tx_pkt.li = 0;
+	tx_pkt.vn = SNTP_VERSION_NUMBER;
+	tx_pkt.mode = SNTP_MODE_CLIENT;
+	int64_t ts_us = k_ticks_to_us_near64(k_uptime_ticks());
```

Comment:
Please also check the compliance results - for instance, this declaration could be moved to the top of the block.